homeassistant/components/bemfa/mqtt.py

```diff
@@ -0,0 +1,110 @@
+"""Support for bemfa service."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import paho.mqtt.client as mqtt
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    EVENT_HOMEASSISTANT_STARTED,
+    EVENT_STATE_CHANGED,
+)
+from homeassistant.core import CoreState, HomeAssistant
+
+from .const import MQTT_HOST, MQTT_KEEPALIVE, MQTT_PORT, TOPIC_PUBLISH
+from .helper import generate_msg, generate_msg_list, generate_topic, resolve_msg
+
+_LOGGING = logging.getLogger(__name__)
+
+
+class BemfaMqtt:
+    """Representation of bemfa."""
+
+    _topic_to_entity_id: dict[str, str] = {}
+    _remove_listener: Any
+    _entity_ids: list[str]
+
+    def __init__(self, hass: HomeAssistant, uid: str, entity_ids: list[str]) -> None:
+        """Initialize."""
+        self._hass = hass
+        self._uid = uid
+        self._entity_ids = entity_ids
+
+        # Init MQTT connection
+        self._mqttc = mqtt.Client(uid, mqtt.MQTTv311)
+        self._mqttc.connect(MQTT_HOST, port=MQTT_PORT, keepalive=MQTT_KEEPALIVE)
+        self._mqttc.on_message = self._mqtt_on_message
+        self._mqttc.loop_start()
+
+        if self._hass.state == CoreState.running:
+            self._connect(None)
+        else:
+            # for situations when hass restarts
+            self._hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STARTED, self._connect)
+
+    def _connect(self, _event) -> None:
+        for entity_id in self._entity_ids:
+            state = self._hass.states.get(entity_id)
+            if state is None:
+                continue
+            topic = generate_topic(state.domain, entity_id)
+            self._topic_to_entity_id[topic] = entity_id
+            self._mqttc.publish(
+                TOPIC_PUBLISH.format(topic=topic),
+                generate_msg(state.domain, state.state, state.attributes),
+            )
+            self._mqttc.subscribe(topic, 2)
+
+        # Listen for state changes
+        self._remove_listener = self._hass.bus.async_listen(
+            EVENT_STATE_CHANGED, self._state_listener
+        )
```

Comment:
We can't make async calls from the MQTT client's thread.

Reply:
Resolved. Thanks!